app/ante/vesting.go

```diff
@@ -109,12 +109,20 @@ func (vdd VestingDelegationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 				)
 			}
 
-			clawbackAccount, isPeriodicVesting := acc.(*vestingtypes.ClawbackVestingAccount)
-			if !isPeriodicVesting {
+			clawbackAccount, isClawback := acc.(*vestingtypes.ClawbackVestingAccount)
+			if !isClawback {
 				// continue to next decorator as this logic only applies to vesting
 				return next(ctx, tx, simulate)
 			}
 
+			// Error if account has locked coins (before surpassing all lockup periods)
+			islocked := clawbackAccount.HasLockedCoins(ctx.BlockTime())
+			if islocked {
+				return ctx, sdkerrors.Wrapf(vestingtypes.ErrVestingLockup,
+					"cannot delegate coins with clawback vesting account, that has locked coins: %s", clawbackAccount.GetLockedOnly(ctx.BlockTime()),
+				)
+			}
```

Comment:
ok, so based on discussion with @khoslaventures, we need to remove this and only enforce the restriction contractually